programs/programs/policyengine/calculators/dependencies/member.py

```diff
@@ -100,6 +100,42 @@ def value(self):
         return self.member.disabled or self.member.long_term_disability
 
 
+class IsUSDADisabledDependency(Member):
+    field = "is_usda_disabled"
+
+    def value(self):
+        return self.member.has_disability()
+
+
+class IsUSDAElderlyDependency(Member):
+    field = "is_usda_elderly"
+
+    def value(self):
+        return self.member.is_elderly()
+
+
+# In this class a dynamic attribute called '_medical_expense_assigned_member_id' is used
+# to ensure that medical expenses are only counted once per household, even when multiple
+# elderly or disabled members are in the same household. This is done because the Member
+# class runs once per each household member and makes it so that the PolicyEngine aggregate
+# the medical expenses across all members when calculating SNAP benefits.
+class MedicalExpenseDependency(Member):
+    field = "medical_out_of_pocket_expenses"
+
+    def value(self):
+        if self.member.is_elderly() or self.member.has_disability():
+            assigned_member_id = getattr(self.screen, "_medical_expense_assigned_member_id", None)
+            if assigned_member_id is None:
+                self.screen._medical_expense_assigned_member_id = self.member.id
+                return self.screen.calc_expenses("yearly", ["medical"])
+            elif assigned_member_id == self.member.id:
+                return self.screen.calc_expenses("yearly", ["medical"])
+            else:
+                return 0
+        else:
+            return 0
```

Comment:
This will not work because the medical expense deduction could be attached to someone who is not elderly or disabled. What you need to do is `self.screen.calc_expenses("yearly", ["medical"]) / count_of_elderly_or_disabled_members`. I assume that using the count of elderly or disabled members is the most accurate assumption to make because those are the members of the household where this expense matters for SNAP purposes. You could also divide the medical expenses by all members, or by the number of adults in the household.

Reply:
In the SNAP [screener](https://www.snapscreener.com/screener/north-carolina), the 'out of pocket medical expenses' option only comes up when the option of to include a disabled/elderly person is enabled. So I think the medical expense bill only counts whenever a household has an elderly or disabled person?

> What you need to do is `self.screen.calc_expenses("yearly", ["medical"]) / count_of_elderly_or_disabled_members`. I assume that using the count of elderly or disabled members is the most accurate assumption to make because those are the members of the household where this expense matters for SNAP purposes.

The if condition essentially does that. If there is at least one disabled or elderly member, then the medical expense is sent to PE.